File: mobile/shared/src/commonMain/kotlin/com/bissbilanz/model/Weight.kt
package com.bissbilanz.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class WeightEntry(
    val id: String,
    val userId: String,
    val weightKg: Double,
    val entryDate: String,
    val loggedAt: String? = null,
    val notes: String? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null,
)

@Serializable
data class WeightEntriesResponse(
    val entries: List<WeightEntry>,
)

@Serializable
data class WeightEntryResponse(
    val entry: WeightEntry,
)

@Serializable
data class WeightTrendEntry(
    @SerialName("entry_date") val entryDate: String,
    @SerialName("weight_kg") val weightKg: Double,
    @SerialName("moving_avg") val movingAvg: Double? = null,
)

@Serializable
data class WeightTrendResponse(
    val data: List<WeightTrendEntry>,
)

File: src/lib/server/validation/responses/weight.ts
import 'zod-openapi';
import { z } from 'zod';

const weightEntrySchema = z
	.object({
		id: z.string().uuid(),
		userId: z.string().uuid(),
		weightKg: z.number(),
		entryDate: z.string(),
		loggedAt: z.string().optional(),
		notes: z.string().nullable(),
		createdAt: z.string().optional(),
		updatedAt: z.string().optional()
	})
	.meta({ id: 'WeightEntry' });

const trendEntrySchema = z
	.object({
		entry_date: z.string(),
		weight_kg: z.number(),
		moving_avg: z.number()
	})
	.meta({ id: 'WeightTrendEntry' });

export const weightEntriesResponseSchema = z
	.object({
		entries: z.array(weightEntrySchema)
	})
	.meta({ id: 'WeightEntriesResponse' });

export const weightEntryResponseSchema = z
	.object({
		entry: weightEntrySchema
	})
	.meta({ id: 'WeightEntryResponse' });

export const weightLatestResponseSchema = z
	.object({
		entry: weightEntrySchema.nullable()
	})
	.meta({ id: 'WeightLatestResponse' });

export const weightTrendResponseSchema = z
	.object({
		data: z.array(trendEntrySchema)
	})
	.meta({ id: 'WeightTrendResponse' });
